components/provisioner/internal/installation/resource_cleanup.go

```diff
@@ -10,36 +10,68 @@ import (
 	"github.com/kubernetes-sigs/service-catalog/pkg/util"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	apiServBeta "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	apiErrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/rest"
 )
 
+const SystemBrokerCRDName = "clusterservicebrokers.servicecatalog.k8s.io"
+const SystemCatalogCRDName = "clusterserviceclasses.servicecatalog.k8s.io"
+const SystemInstanceCRD = "serviceinstances.servicecatalog.k8s.io"
+
 const (
 	ClusterServiceBrokerNameLabel   = "servicecatalog.k8s.io/spec.clusterServiceBrokerName"
 	ClusterServiceClassRefNameLabel = "servicecatalog.k8s.io/spec.clusterServiceClassRef.name"
 )
 
-type ServiceCatalogClient interface {
+type CleanupClient interface {
 	PerformCleanup(resourceSelector string) error
 }
 
-func NewServiceCatalogClient(kubeconfig *rest.Config) (ServiceCatalogClient, error) {
+type CrdsManager interface {
+	List(ctx context.Context, opts metav1.ListOptions) (*apiServBeta.CustomResourceDefinitionList, error)
+}
+
+func NewServiceCatalogCleanupClient(kubeconfig *rest.Config) (CleanupClient, error) {
 	scCli, err := sc.NewForConfig(kubeconfig)
 	if err != nil {
 		return &serviceCatalogClient{}, err
 	}
 
-	return &serviceCatalogClient{client: scCli}, nil
+	apiExtensionsClientSet, err := apiextensionsclient.NewForConfig(kubeconfig)
+
+	if err != nil {
+		return &serviceCatalogClient{}, err
+	}
+
+	crdsInterface := apiExtensionsClientSet.ApiextensionsV1beta1().CustomResourceDefinitions()
+
+	return &serviceCatalogClient{
+		client:      scCli,
+		crdsManager: crdsInterface,
+	}, nil
 }
 
 type serviceCatalogClient struct {
-	client sc.Interface
+	client      sc.Interface
+	crdsManager CrdsManager
 }
 
 func (s *serviceCatalogClient) PerformCleanup(resourceSelector string) error {
+	exist, err := s.ensureCRDsExist()
+	if err != nil {
+		return errors.Wrapf(err, "while checking CustomResourceDefinitions")
+	}
+
+	if !exist {
+		logrus.Info("Service Catalog not installed properly. Cleanup skipped")
```

Comment:
The information in the log file doesn't contain RuntimeID. Please consider adding it.